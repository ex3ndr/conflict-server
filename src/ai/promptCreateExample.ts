import Handlebars from "handlebars";
import { prompt_conflict_example } from "./prompts/_prompts";

const template = Handlebars.compile(prompt_conflict_example);

export function promptCreateExample(args: {
    nameA: string,
    nameB: string
}) {
    return template(args);
}