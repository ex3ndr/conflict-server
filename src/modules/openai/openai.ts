import OpenAI from 'openai';

export const openai = new OpenAI({
    apiKey: process.env.OPENAI_TOKEN
});

export async function executeOpenAI(messages: { role: 'user' | 'assistant' | 'system', content: string }[]) {
    let res = await openai.chat.completions.create({
        model: 'gpt-3.5-turbo',
        messages: messages
    });
    return res.choices[0].message.content!;
}