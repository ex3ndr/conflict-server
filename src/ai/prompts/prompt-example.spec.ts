import { executeOpenAI } from "../../modules/openai/openai";
import { promptCreateExample } from "../promptCreateExample";

describe('prompt-example', () => {
    jest.setTimeout(60000);
    it('should create an example', async () => {
        jest.setTimeout(25000);
        let nameA = 'Cat';
        let nameB = 'Dog';
        let initialMessage = promptCreateExample({ nameA, nameB });
        let executed = await executeOpenAI([{
            role: 'system',
            content: initialMessage
        }]);
        expect(executed).toMatchSnapshot();
    });
});