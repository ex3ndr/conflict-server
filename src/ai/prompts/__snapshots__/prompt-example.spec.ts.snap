// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`prompt-example should create an example 1`] = `""Cat and Dog are arguing over who gets the last piece of pizza.""`;
